Dart/src/com/jetbrains/lang/dart/ide/formatter/DartSpacingProcessor.java

```diff
@@ -404,9 +409,47 @@ private Spacing setBraceSpace(boolean needSpaceSetting,
       return Spacing.createDependentLFSpacing(spaces, spaces, textRange, mySettings.KEEP_LINE_BREAKS, mySettings.KEEP_BLANK_LINES_IN_CODE);
     }
     else {
-      final int lineBreaks = braceStyleSetting == CommonCodeStyleSettings.END_OF_LINE ||
-                             braceStyleSetting == CommonCodeStyleSettings.NEXT_LINE_IF_WRAPPED ? 0 : 1;
+      final int lineBreaks =
+        braceStyleSetting == CommonCodeStyleSettings.END_OF_LINE || braceStyleSetting == CommonCodeStyleSettings.NEXT_LINE_IF_WRAPPED
+        ? 0
+        : 1;
       return Spacing.createSpacing(spaces, spaces, lineBreaks, false, 0);
     }
   }
+
+  private boolean allCascadesAreSameMethod(ASTNode[] children) {
+    for (int i = 1; i < children.length; i++) {
+      if (!cascadesAreSameMethod(children[i - 1], children[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean cascadesAreSameMethod(ASTNode child1, ASTNode child2) {
+    ASTNode call1 = child1.getLastChildNode();
+    if (call1.getElementType() == CALL_EXPRESSION) {
+      ASTNode call2 = child2.getLastChildNode();
+      if (call2.getElementType() == CALL_EXPRESSION) {
+        String name1 = getImmediateCallName(call1);
+        if (name1 != null) {
+          String name2 = getImmediateCallName(call2);
+          if (name1.equals(name2)) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+
+  private String getImmediateCallName(ASTNode callNode) {
```

Comment:
I haven't trained myself to recognize warnings yet, I guess. There is one, I hadn't noticed it, and I don't even agree that what it is saying is good practice. But I'll change it anyway.